### PR TITLE
Detect jetpack add-on (close #192)

### LINKED ISSUE #192: Detect JPM addons
Needed to provide more metadata.

### PR DESCRIPTION
**Pull request by tofumatt**
Closes #230 as well; this is essentially a second pass at that.

### CODE REVIEW

File: tests/test.validator.js [hunk 1]
```diff
@@ -125,7 +125,7 @@ describe('Validator', function() {
   });
 
   it('should throw when message.type is undefined', () => {
-    var addonValidator = new Validator({_: ['tests/example.xpi']});
+    var addonValidator = new Validator({_: ['tests/fixtures/example.xpi']});
```
Comment (muffinresearch) [inline tests/test.validator.js]: Nice, I like this path change :+1:

File: src/validator.js [hunk 2]
```diff
@@ -36,6 +37,7 @@ export default class Validator {
     this.chalk = new chalk.constructor(
       {enabled: !this.config.boring});
     this.collector = new Collector();
+    this.addonMetaData = {};
```
Comment (muffinresearch) [inline src/validator.js]: Good idea.

File: src/validator.js [hunk 3]
```diff
@@ -190,8 +192,13 @@ export default class Validator {
   }
 
   getAddonMetaData() {
+    var _xpiMeta;
+
     return this.xpi.getMetaData()
       .then((xpiMeta) => {
+        // Used later to get add-on architecture.
+        _xpiMeta = xpiMeta;
```
Comment (muffinresearch) [inline src/validator.js]: This will need a rename - but I'm sure you know that already :)

File: src/validator.js [hunk 4]
```diff
@@ -420,4 +429,19 @@ export default class Validator {
       return this.scan(_Xpi);
     }
   }
+
+  _getAddonArchitecture(xpiMetaData) {
+    // If we find a file named bootstrap.js this is assumed to be a
+    // Jetpack add-on: https://github.com/mozilla/amo-validator/blob/7a8011aba8bf8c665aef2b51eb26d0697b3e19c3/validator/testcases/jetpack.py#L154
```
Comment (muffinresearch) [inline src/validator.js]: Does eslint/jscs not whine about the length of this line?

File: src/validator.js (hunk 2 above)
Comment (muffinresearch) [inline src/validator.js]: I wonder if this should be null, then we could more easily do a check and return it if it's already defined. Can be a todo or new issue.

**Comment by muffinresearch** [comment]
Going to need to renames following #251 but otherwise r+wc

File: src/validator.js (hunk 4 above)
Comment (tofumatt) [inline src/validator.js]: Nope. It's smart enough to realise it's a URL and not complain.

- tofumatt (Sent from mobile)

Comment (muffinresearch) [inline src/validator.js]: Oh shiny. Cool

File: src/validator.js (hunk 2 above)
Comment (tofumatt) [inline src/validator.js]: I can integrate that into this PR no problem. Have to do a rebase anyway after #251 lands :smiley:

Comment (tofumatt) [inline src/validator.js]: Actually, it involves extra tests and such so I'll make a new issue for it, just to keep this lean. #257

**Pull request merged**